src/Symfony/Component/Validator/Mapping/PropertyMetadata.php

```diff
@@ -50,8 +50,21 @@ public function getPropertyValue($object)
     {
         $reflProperty = $this->getReflectionMember($object);
 
-        if (\PHP_VERSION_ID >= 70400 && !$reflProperty->isInitialized($object)) {
-            return null;
+        if (\PHP_VERSION_ID >= 70400 && $reflProperty->hasType() && !$reflProperty->isInitialized($object)) {
+            // There is no way to check if a property has been unset or it is uninitialized.
+            // When trying to access an uninitialized property, __get method is triggered in PHP 7.4.0, but not in 7.4.1+.
```

Comment:
Does this mean that there was a bug in PHP 7.4.0 and `__get()` should never have been called? In this case, I am not convinced that we should do anything here as the behaviour is inline with what happens in PHP 7.4.1+.

Reply:
see also https://3v4l.org/8E83P

Reply:
or should we call `__get()` directly?

Reply:
There is a discussion [PHP Bug # 78904] (https://bugs.php.net/bug.php?id=78904).

This is a PHP related problem, I think. Framework should not implement any hack for this. ProxyManager and others are currently not compatible with PHP 7.4.1+.

Reply:
`__get` **must** be called when accessing a property, if the property has been explicitly unset, but not if it is uninitialized.
On the other side `ReflectionProperty::isInitialized` is returning false for both situations and there is no *safe* way to distinguish one state from the other.

Reply:
@greedyivan The problem is not limited to the ProxyManager and I don't think that symfony could stop supporting PHP 7.4.1+ until the discussion on PHP is resolved.

Reply:
Here is the case:
```php
class Foo
{
    public string $unintialized;
    public string $unsetted;
    
    public function __construct() {
        unset($this->unsetted);
    }

    public function __get($var)
      {
        return 'not set';
     }
}

$foo = new Foo();

assert(null === $foo->unintialized) // validator should return null here
assert('not set' === $foo->unsetted); // validator should return the actual value
```

Reply:
@greedyivan It is exactly what this patch does

Reply:
@alekitto, agree. I've just check it out.

Only two thinks:
- `$reflProperty->hasType()` means nothing here
- `} catch (\TypeError $e) {` - here must be `\Error`

Reply:
The `hasType` has been added as if the property is not typed, no additional check should be made.
I think that `\Error` is too generic. Per typed properties RFC `\TypeError` is raised in case of access an uninitialized non-nullable typed property.

Reply:
> The `hasType` has been added as if the property is not typed, no additional check should be made.
We have to do some check:
- for non-typed in this case we check `hasType`
- for typed we check `hasType` and `isInitialized`
Using only one check `isInitialized` does the same and has no additional cost.

> I think that `\Error` is too generic. Per typed properties RFC `\TypeError` is raised in case of access an uninitialized non-nullable typed property.
https://3v4l.org/lmTYH

Don't trust docs blindly.

Reply:
> https://3v4l.org/lmTYH
According that, additional tests should be added.

- typed uninitialized propery with no unset in constructor and with __get method.

This test would validate a correct way of `\Error` catching. In case if it will be changed to `\TypeError` in future.

Reply:
I've just added a check. Too bad that RFC is not respected about the `\TypeError`

Reply:
> Too bad that RFC is not respected about the `\TypeError`

[Doc Bug #79206](https://bugs.php.net/bug.php?id=79206)
Let's see, it is a doc bug or not.